lalrpop/src/grammar/parse_tree.rs

```diff
@@ -710,6 +718,55 @@ impl NonterminalData {
     }
 }
 
+impl Attribute {
+    /// get the (key, value) of an attribute of the form #[key = "value"]
+    ///
+    /// # Panics
+    ///
+    /// panics if the attribute is not an equal, i.e. #[attr] or #[attr(...)]
+    pub fn unwrap_arg_equal(self) -> (Atom, String) {
+        match self.arg {
+            AttributeArg::Paren(arg) => {
+                let attr = arg.into_iter().next().unwrap();
+                (attr.id, attr.arg.unwrap_equal())
+            }
+            _ => panic!("not an attribute of type attr(arg = value)"),
+        }
+    }
+
+    /// get the (key, value) of an attribute of the form #[key = "value"]
+    pub fn get_arg_equal(&self) -> Option<(&Atom, &String)> {
+        match &self.arg {
+            AttributeArg::Paren(arg) => arg
+                .first()
+                .and_then(|attr| attr.arg.get_equal().map(|eq| (&attr.id, eq))),
+            _ => None,
+        }
+    }
+}
```

Comment:
Thanks for the documentation. These two functions seem a little redundant? Do we need both or can we get away with the `get` version and just unwrap as needed?

Reply:
yeah looking at it, they definetly look redundant and somehat hacky.
There's two of them because the unwrap one owns self. but refactoring is definitely possible.